### DISCUSSION TITLE
Web Gui Cert

### DISCUSSION THREAD

**Discussion by ng1991** [Security-Onion-Solutions/securityonion#9933] [Unsupported Versions]
Hi All,

Hopefully this isnt a big issue but I followed this guide:

https://nano.dannyvacar.ca/post/custom-ssl-certificate-for-security-onion-web-ui/

Everything seemed to work fine and I cold see my custom cert on the web gui and everything seemed to be working however after rebooting a sensor I noticed that the docker would not start, on investigation I was getting this error:

{"caller":"request_queries.go:162","err":"rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","method":"RequestQueries","reauth":false,"res":"null","severity":"info","took":"88.416µs","ts":"2023-03-15T12:36:18.044081225Z","uuid":"42c7a214-8afd-46f1-8daa-96af54e34e4c"}

I then added the Root and SubCa to the node via:

added certs in this directory "/etc/pki/ca-trust/source/anchors/"
then ran "update-ca-trust extract"

After this the docker started and again it looked good but I was still seeing the error in my /var/log/messages.

For now I have reverted back to the default self signed cert and the error has cleared from the manager and the sensors but I was wondering on the best way to add a custom cert so I dont get the SSL error when accessing the manager.

Regards

NG

**Comment by InfosecGoon** [comment]
All you should need to do is replace the managerssl.crt and .key files in /etc/pki, as outlined in the link. That certificate pair is specific to the web interface and isn't used for any internode communication - I think you might have had two separate issues occurring at the same time.

**Reply by ng1991** [reply]
Hi thanks for your response.

I just added the certs back and as soon as I restarted nginix container the errorrs started appearing in /var/log/messages

![image](https://user-images.githubusercontent.com/115169428/225641382-69f01702-bfbf-455c-ae84-169c4e937cbc.png)

Everything does seem to be working from a node to manger perspective but the logs are just constantly throwing the " x509: certificate signed by unknown authority\"","method":"RequestQueries","reauth":false,"res":"null","severity":"info"," error on both the manager and forward nodes.

If I issue back the default self signed the errors don't appear.

I also restarted a node after the updating od the certs and docker never initializes
![image](https://user-images.githubusercontent.com/115169428/225644243-12a179b1-ba57-47e7-aec2-e2c67c82cc19.png)

However after adding the sub and root and rebooting it then starts but still see the errors in the log so does appear to be using them certs elsewhere but not sure where.
![image](https://user-images.githubusercontent.com/115169428/225648631-75bfafa4-20a4-4caf-aba8-99ed2189fc0d.png)

**Reply by ng1991** [reply]
I think its related to the osquery, we dont use this part of the system but moitoring the log after reverting the certs thats what seems to start working and the errors clear.

I have seen this mentioned in some other posts indicating I may need to add something to intca.crt? is that correct? Or even the /etc/so-launcher/root.pem?

Below shows the error and then what happened after reverting the cert
![image](https://user-images.githubusercontent.com/115169428/225655174-a790a097-8e86-43f0-baa2-0fe16ab7166c.png)

**Reply by InfosecGoon** [reply]
You're right, those errors are coming from launcher, which is a wrapper around the osquery agent.

**Reply by ng1991** [reply]
Thanks for confirming I appreciate the help from this forum.

So do you know how I would stop these errors? Like I say we don't currently use the osquery  function but is it just a case of updating the intca.cer as shown in other guides?

**Reply by ng1991** [reply]
Hi,

Just wondering if you could point me in the right direction on this so I can apply the gui certs and not have the errors?

**Reply by InfosecGoon** [reply]
If you don't use the osquery function, you can remove the launcher package with "sudo rpm -e launcher-final".

If you'd like to reinstall it later, you can regenerate the install packages on the download page with "sudo salt-call state.apply fleet.event_gen-packages" and then manually install the new one.

**Reply by ng1991** [reply]
Thanks will have a think 

Cheers
